code/script.js

```diff
@@ -1,10 +1,270 @@
 // All the DOM selectors stored as short variables
-const chat = document.getElementById('chat')
+const chat = document.getElementById('chat');
+const form = document.getElementById('name-form');
+const inputWrapper = document.getElementById('input-wrapper');
+const nameInput = document.getElementById('name-input');
 
 // Global variables, if you need any, declared here
 
+let answer = 1;
+
 // Functions declared here
 
+const handleNameInput = (event) => {
+  event.preventDefault()
+  const name = nameInput.value;
+  showMessage(name, 'user');
+  nameInput.value = "";
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage(`Hi ${name}! I am here to assist you booking a trip.`, 'bot'),1000);
+  setTimeout(() => askWhereTo(),2500);
+}
+
+const askWhereTo = () => {
+  showMessage("Are you planning a trip within Europe or outside Europe?","bot");
+
+  inputWrapper.innerHTML=`
+  <button id="button-europe">
+    Within Europe
+  </button>
+  <button id="button-outside">
+    Outside Europe
+  </button>
+  `
+  document
+  .getElementById("button-europe")
+  .addEventListener("click", () => {
+  showMessage("Within Europe", "user")
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage("Wonderful! There is so much to explore in Europe.", "bot"), 1500);
+  setTimeout(() => showMessage("We have a great variety of destinations at the moment.", "bot"),3000);
+  setTimeout(() => askType(), 5000);
+  })
+
+  document
+  .getElementById("button-outside")
+  .addEventListener("click", () => {
+  showMessage("Outside Europe", "user")
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage("Great, so somewhere far then!", "bot"),1500);
+  setTimeout(() => showMessage("We have a great variety of destinations at the moment.", "bot"),3000);
+  setTimeout(() => askType(), 5000);
+  })
+}
```

Comment:
This code structure is repeated a few times what is the ideal situation to create a more generic function to handle this task every time 🤓 Please try to do it 😸 If you will you can always ask me for more tips 😸